linux/example/flutter_embedder_example.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <cstdlib>
+#include <iostream>
+
+#include <flutter_embedder.h>
+
+int main(int argc, char **argv) {
+  // Edit this as needed.
+  std::string flutter_example_root = "../../flutter/examples/flutter_gallery";
+  if (!glfwInit()) {
+    std::cout << "Couldn't init GLFW";
+  }
+  auto window = CreateFlutterWindow(
+      640, 480, flutter_example_root + "/lib/main.dart",
```

Comment:
Let's use snapshot mode rather than source mode for the example.

Reply:
I'll move it to that, but for now I'm keeping it here since snapshot mode is giving me errors on my machine.  I'll update once I've figured out what's going on.

Reply:
Okay, it looks like we need to run `flutter build flx --local-engine-src-path=/path/to/eng --local-engine host_debug_unopt` in order for snapshot mode to run properly (at least on my machine).

Reply:
I can see this causing similar issues for the MacOS build since it just uses `../.../flutter/bin/flutter build flx` with no reference to the user's local engine build.